data-collection/data-collection/View Controllers/Rich Popup View Controllers/Rich Popup View Controller/RichPopupSelectRelatedRecordViewController/RichPopupSelectRelatedRecordViewController.swift

```diff
@@ -0,0 +1,60 @@
+//// Copyright 2019 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import ArcGIS
+import UIKit
+
+protocol RichPopupSelectRelatedRecordViewControllerDelegate: AnyObject {
+    func richPopupSelectRelatedRecordViewController(_ richPopupSelectRelatedRecordViewController: RichPopupSelectRelatedRecordViewController, didSelectPopup popup: AGSPopup)
+}
+
+class RichPopupSelectRelatedRecordViewController: UITableViewController {
+    
+    var popups: [AGSPopup]!
+    
+    var currentRelatedPopup: AGSPopup?
+    
+    weak var delegate: RichPopupSelectRelatedRecordViewControllerDelegate?
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        
+        if let current = currentRelatedPopup, let index = popups.firstIndex(where: { $0 == current }) {
```

Comment:
I think I already commented on this line, but I completely missed something. That closure isn't even needed.

```suggestion
        if let current = currentRelatedPopup, let index = popups.firstIndex(of: current) {
```

I've been looking for ways to use `Optional.map(_:)` and `Optional.flatMap(_:)`. You could map the index:

```swift
if let current = currentRelatedPopup, let indexPath = popups.firstIndex(of: current).map({ IndexPath(row: $0, section: 0) }) {
    tableView.selectRow(at: indexPath, animated: false, scrollPosition: .middle)
}
```

Or you could flat map the current related popup:

```swift
if let index = currentRelatedPopup.flatMap({ popups.firstIndex(of: $0) }) {
    let indexPath = IndexPath(row: index, section: 0)
    tableView.selectRow(at: indexPath, animated: false, scrollPosition: .middle)
}
```

I personally kind of like the `flatMap` version. You don't have to use either of those if you don't want to.